app/src/androidTest/java/org/mozilla/fenix/ui/robots/AddonDetailedMenuRobot.kt

```diff
@@ -0,0 +1,56 @@
+package org.mozilla.fenix.ui.robots
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import org.hamcrest.CoreMatchers.allOf
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.click
+
+class AddonDetailedMenuRobot {
+
+    fun verifyCurrentAddonMenu() = assertAddonMenuItems()
+
+    class Transition {
+        fun goBack(interact: SettingsSubMenuAddonsManagerRobot.() -> Unit): SettingsSubMenuAddonsManagerRobot.Transition {
+            fun goBackButton() = onView(allOf(withContentDescription("Navigate up")))
+            goBackButton().click()
+
+            SettingsSubMenuAddonsManagerRobot().interact()
+            return SettingsSubMenuAddonsManagerRobot.Transition()
+        }
+
+        fun removeAddon(interact: SettingsSubMenuAddonsManagerRobot.() -> Unit): SettingsSubMenuAddonsManagerRobot.Transition {
+            removeAddonButton().click()
+
+            SettingsSubMenuAddonsManagerRobot().interact()
+            return SettingsSubMenuAddonsManagerRobot.Transition()
+        }
+    }
+
+    private fun assertAddonMenuItems() {
+        enableSwitchButton().check(matches(isCompletelyDisplayed()))
+        settingsButton().check(matches(isCompletelyDisplayed()))
+        detailsButton().check(matches(isCompletelyDisplayed()))
+        permissionsButton().check(matches(isCompletelyDisplayed()))
+        removeAddonButton().check(matches(isCompletelyDisplayed()))
+    }
+}
+
+private fun enableSwitchButton() =
+    onView(withId(R.id.enable_switch))
+
+private fun settingsButton() =
+    onView(withId(R.id.settings))
+
+private fun detailsButton() =
+    onView(withId(R.id.details))
+
+private fun permissionsButton() =
+    onView(withId(R.id.permissions))
+
+private fun removeAddonButton() =
+    onView(withId(R.id.permissions))
```

Comment:
Is this a dynamic button with the same ID 'R.id.permissions'?

Reply:
Yup, the id is reused on this screen